azurerm/internal/services/apimanagement/api_management_api_resource.go

```diff
@@ -494,3 +568,52 @@ func flattenApiManagementApiSubscriptionKeyParamNames(paramNames *apimanagement.
 
 	return []interface{}{result}
 }
+
+func expandApiManagementOpenIDAuthenticationSettingsBearerTokenSendingMethods(input interface{}) *[]apimanagement.BearerTokenSendingMethods {
+	if input == nil {
+		return nil
+	}
+	results := make([]apimanagement.BearerTokenSendingMethods, 0)
+
+	vs := input.(*schema.Set).List()
+	for _, v := range vs {
+		results = append(results, apimanagement.BearerTokenSendingMethods(v.(string)))
+	}
+
+	return &results
+}
+
+func flattenApiManagementOAuth2Authorization(input *apimanagement.OAuth2AuthenticationSettingsContract) []interface{} {
+	if input == nil {
+		return make([]interface{}, 0)
+	}
+
+	result := make(map[string]interface{})
+
+	result["authorization_server_name"] = *input.AuthorizationServerID
```

Comment:
We should check this for nil before setting:
```suggestion
	authServerId := ""
	if input.AuthorizationServerID != nil {
		authServerId = *input.AuthorizationServerID
	}
	result["authorization_server_name"] = authServerId
```